src/js/components/form/FormGroupHeading.js

```diff
@@ -0,0 +1,56 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import FormGroupHeadingContent from './FormGroupHeadingContent';
+
+function getChildren(children, required) {
+  if (!required) {
+    return children;
+  }
+
+  // We add the required field indicator after the first child.
+  return React.Children.toArray(children).reduce(
+    function (accumulator, child, index) {
+      accumulator.push(child);
+
+      if (index === 0) {
+        accumulator.push(
+          <FormGroupHeadingContent
+            className="text-danger"
+            key={`${index}-asterisk`}>
+            *
+          </FormGroupHeadingContent>
+        );
+      }
+
+      return accumulator;
+    },
+    []
+  );
+}
+
+function FormGroupHeading({className, children, required}) {
+  const classes = classNames('form-group-heading', className);
+
+  return (
+    <div className={classes}>
+      {getChildren(children, required)}
```

Comment:
I could have mixed up the terms, what I really meant was using boolean arguments to control the internal flow of a function. It's not really a problem but rather a thing that one should avoid. Because it's hard to reason about such functions.

The question is: why don't you extract this argument into an `if` conditional and call that function conditionally rather then running it always and conditionally getting a result :) 

How is `getChildren(children, true)` different from `getChildren(children, false)` or `getChildren(children, isRequired)`? You can't tell without looking into the function.
But something like `injectAsteriskNode(children)` immediately tells you what's going to happen.

There's a good short article: https://silkandspinach.net/2004/07/15/avoid-boolean-parameters/

Please note it is different from a React function that internally checks `this.state.arg` or `this.props` because in this case you're using an internal state of a React component. But even then I'd rather put this control logic on the upper level.

Feel free to bring this topic to one of our sync meetings so we could discuss it more.